### PR TITLE
Fix issue on cookie and create new Pages

### PR DESCRIPTION
**Pull request by HenryChen233**
- Create new pages for wrong password handling
- Fixed problem on getting cookie from the browser
- Fix problems on the UI and create new pages for new function.

### CODE REVIEW

File: request_handler.go [hunk 1]
```diff
@@ -24,100 +25,51 @@ type Profile struct {
...
 	return &pageInfo, err
 }
 
-//Function to read public information without login and render the web pages
+// Function to read public information without login and render the web pages
 func readFromPublic(username string) (*PublicInfo, error) {
 	link := "http://localhost:8080/v1/accounts/" + username
 	resp, err := http.Get(link)
 	if err != nil {
 		log.Fatal(err)
```
Comment (sleepingpig) [inline request_handler.go]: Shouldn't use `log.Fatal` since it will terminate the server.

File: request_handler.go [hunk 2]
```diff
@@ -149,36 +102,57 @@ func editHandler(w http.ResponseWriter, r *http.Request, title string) {
...
-
 func saveHandler(w http.ResponseWriter, r *http.Request, title string) {
 	email := r.FormValue("email")
 	description := r.FormValue("description")
 	originalPage, err := readFromBackend(title)
 	if err != nil {
 		log.Println(err)
 	}
+	// Get cookie from browser
+	cookie, err := r.Cookie("name-cookie")
```
Comment (sleepingpig) [inline request_handler.go]: `name-cookie`? What's this?

File: request_handler.go [hunk 3]
```diff
@@ -149,36 +102,57 @@ func editHandler(w http.ResponseWriter, r *http.Request, title string) {
...
+	updateInfo.Description = description
+	// Json format transformation
+	jsonData, err := json.Marshal(updateInfo)
+	if err != nil {
+		log.Println(err)
+	}
+	Data := string(jsonData)
+	payload := strings.NewReader(Data)
+	// Send http request
+	link := "http://localhost:8080/v1/accounts/@me?token=" + cookie.Value
```
Comment (sleepingpig) [inline request_handler.go]: Why you are appending cookie to the url?

File: request_handler.go [hunk 4]
```diff
@@ -199,97 +184,129 @@ func createHandler(w http.ResponseWriter, r *http.Request, title string){
...
 	user.Password = password
 	userData, err := json.Marshal(user)
 	userString := string(userData)
 	payload := strings.NewReader(userString)
 	response, err := http.Post("http://localhost:8080/v1/sessions/", "application/json", payload)
+	if response.StatusCode == 401 {
+		http.Redirect(w, r, "/loginError/", http.StatusFound)
+	}
 	if err != nil {
 		log.Println(err)
```
Comment (sleepingpig) [inline request_handler.go]: It shouldn't just ignore error.

File: request_handler.go [hunk 5]
```diff
@@ -199,97 +184,129 @@ func createHandler(w http.ResponseWriter, r *http.Request, title string){
...
 	}
 	// Get token from login response from backend
-	token = response.Header.Get("Set-Cookie")
-	Cookie := http.Cookie{Name:user.Username,
+	token := response.Header.Get("Set-Cookie")
+	fmt.Println(token)
```
Comment (sleepingpig) [inline request_handler.go]: Shouldn't print the cookie.

File: request_handler.go [hunk 6]
```diff
@@ -199,97 +184,129 @@ func createHandler(w http.ResponseWriter, r *http.Request, title string){
...
 	renderTemplate(w, "register", &p)
 }
 
+//
 func privateHandler(w http.ResponseWriter, r *http.Request, title string) {
-	fmt.Println("Cookie.......")
-	cookie,_ := r.Cookie("Alice")
-	fmt.Println(cookie)
-	link := "http://localhost:8080/v1/accounts/@me?token=" + token
+	//fmt.Println("Cookie.......")
```
Comment (sleepingpig) [inline request_handler.go]: Don't keep commented code.

Comment (HenryChen233) [reply request_handler.go]: Corrected

File: request_handler.go (hunk 2 above)
Comment (HenryChen233) [reply request_handler.go]: Yes the name is not clear enough, how about rename it as `login-cookie`

File: request_handler.go (hunk 1 above)
Comment (HenryChen233) [reply request_handler.go]: Is it okay to correct it to log.Println(err)?

File: request_handler.go (hunk 2 above)
Comment (DandilionLau) [reply request_handler.go]: I suggest setting the name to `session` to sync up with backend.

Comment (HenryChen233) [reply request_handler.go]: Okay, I would correct it.

**Pull request closed without merging**